Add nfsen_base to config_definitions.json

whitej46 opened the pull request:
Added nfsen_base so that nfsen stats feature works.  I had to add as an array as it was the way I was able to make it work, even though another type may make more sense from a user interface perspective.  I put it in order 10 as there seemed to be a gap there, but one could argue that it should be higher and other nfsen items renumbered.

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [ ] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [ ] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

bnerickson commented:
I applied this patch to test and it broke the stats page.  I define nfsen_base in config.php.  In case this patch is approved as-is and someone else has the same problem as me, here's the fix:

(Replace /abc/defwith your nfsen base directory) Instead of :
```
$config['nfsen_base'][] = '/abc/def';
```

Replace with:
```
$config['nfsen_base'] = [];
$config['nfsen_base'][] = '/abc/def';
```

It looks like ./includes/html/functions.inc.php is running a loop over each nfsen_base folder defined in the array.  If the required folder doesn't exist in the correct layout (rrds aren't in /var/nfsen in my case), then it looks like the stats page bombs out.  Resetting the array then adding your own path works.

whitej46 commented:
Thanks.  I wasn't able to get it working using config.php, which is why I went the route of adding it to config_definitions.json.  If there is a way that I can add it that won't break existing config.php settings, please let me know (my understanding of the inner workings is remedial at best).

murrant reviewed (changes requested):
This is missing the text and needs to go in lang/en/settings.php

Also, perhaps the default needs to be just an empty array?

murrant commented:
@bnerickson the error message might be useful.  To my eyes a non-existent directory shouldn't cause an error.

whitej46 commented:
@murrant I updated lang/en/settings.php and changed the default array to be blank (good suggestion - default I used doesn't even match my installation).  I'm believe that my updates are reflected in this pull request, but let me know if I need to do something different to make them accessible.

bnerickson commented:
> @bnerickson the error message might be useful. To my eyes a non-existent directory shouldn't cause an error.

Unfortunately, there is no error message in librenms.log nor in the browser console nor in apache logs (nor SELinux).  The stats just fail to load.

With patch and manual definition in config.php:
<img width="780" alt="with_patch" src="https://github.com/librenms/librenms/assets/4349917/85d7f063-f147-4960-b6f6-4471cfe18402">

Without patch and manual definition in config.php:
<img width="970" alt="without_patch" src="https://github.com/librenms/librenms/assets/4349917/774971a9-ac2a-4a67-a13e-c5cfb2d02e28">

bnerickson commented:
Setting the default to an empty array no longer breaks the stats page on my installation.  Cheers!
